nncf/quantization/algorithms/accuracy_control/algorithm.py

```diff
@@ -0,0 +1,363 @@
+                 max_drop: float = 0.01,
+                 is_native: bool = True):
+        """
+        :param algo_backend: The `AccuracyControlAlgoBackend` algo backend.
+        :param ranking_subset_size: The number of data items that will be selected from
+            the dataset to rank groups of quantizers.
+        :param max_num_iterations: A maximal number of iterations.
+        :param max_drop: The maximum absolute accuracy drop that should be achieved.
+        """
+        self.algo_backend = algo_backend
+        self.ranking_subset_size = ranking_subset_size
+        self.max_num_iterations = max_num_iterations
+        self.max_drop = max_drop
+        # TODO(andrey-churkin): Should be removed when native implementation
+        # will become the main one.
+        self.is_native = is_native
+
+    def restore_accuracy(self,
+                         initial_model: TModel,
+                         initial_metric: float,
+                         quantized_model: TModel,
+                         quantized_metric: float,
+                         validation_dataset: Dataset,
+                         validation_fn: Callable[[Any, Iterable[Any]], float]) -> TModel:
+        """
+        Restores the accuracy of the quantized model by removing the groups of quantizers
+        that contribute the most to the drop in accuracy.
+
+        :param initial_model: Initial model (not quantized).
+        :param initial_metric: Metric value for initial model.
+        :param quantized_model: Quantized model.
+        :param quantized_metric: Metric value for quantized model.
+        :param validation_dataset: A dataset for the validation process.
+        :param validation_fn: A validation function to validate the model. It should take
+            two argumets:
+            - `model`: model to be validate.
+            - `validation_dataset`: dataset that provides data items to
+                validate the provided model.
+            The function should return the value of the metric with the following meaning:
+            A higher value corresponds to better performance of the model.
+        :return: The quantized model whose metric `final_metric` is satisfied the following condition
+
+            initial_metric - final_metric <= max_drop.
+        """
+        accuracy_drop = initial_metric - quantized_metric
+        nncf_logger.info(f'Accuracy drop: {accuracy_drop}')
+
+        if accuracy_drop <= self.max_drop:
+            return quantized_model
+
+        nncf_logger.info('Changing the scope of quantizer nodes.')
+
+        initial_model_graph = NNCFGraphFactory.create(initial_model)
+        quantized_model_graph = NNCFGraphFactory.create(quantized_model)
+
+        # TODO(andrey-churkin): We need to match constant names when the
+        # quantized model was got using POT. For example, we have the
+        # `Constant_63974886249` constant name in the quantized model,
+        # but `Constant_6397` in the initial model.
```

Comment:
We need to check that the described case with node renaming does not apply to the native implementation. I assume, the behavior also applies to nGarph while the model is creating/loading into the plugin.

Reply:
Also, it might be that the current approach for the constants is not quite correct and it's needed to revise in the future because this kind of backend-specific shouldn't be in the algorithm's code.

Reply:
> We need to check that the described case with node renaming does not apply to the native implementation. I assume, the behavior also applies to nGarph while the model is creating/loading into the plugin.

I have verified it. It relates only to POT.